dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeHandler.java

```diff
@@ -209,10 +209,16 @@ public void received(Channel channel, Object message) throws RemotingException {
                     Exception e = new Exception("Dubbo client can not supported string message: " + message + " in channel: " + channel + ", url: " + channel.getUrl());
                     logger.error(e.getMessage(), e);
                 } else {
-                    String echo = handler.telnet(channel, (String) message);
-                    if (echo != null && echo.length() > 0) {
-                        channel.send(echo);
+                    boolean enableTelnet = channel.getUrl().getParameter(Constants.TELNET, true);
+                    if (enableTelnet){
+                        String echo = handler.telnet(channel, (String) message);
+                        if (echo != null && echo.length() > 0) {

```

Comment:
can be replace with
if(StringUtils.isNotEmpty(echo)) {
    channel.send(echo);
}

Reply:
yes,StringUtils.isNotEmpty is more cleaner as it has the same meaning as ```echo != null && echo.length() > 0```

Reply:
Thanks.